specs/MultiProfile.md

```diff
@@ -169,20 +169,37 @@ void ScenarioCookieManagement::DeleteAllCookies()
 ### Delete profile
 
 ```cpp
-HRESULT AppWindow::DeleteProfile(ICoreWebView2Controller* controller)
+HRESULT AppWindow::DeleteProfile(ICoreWebView2* webView2)
 {
-    wil::com_ptr<ICoreWebView2> coreWebView2;
-    CHECK_FAILURE(controller->get_CoreWebView2(&coreWebView2));
-    auto webview7 = coreWebView2.try_query<ICoreWebView2_7>();
-    if (webview7)
+    DCHECK(webView2);
+    wil::com_ptr<ICoreWebView2Profile> profile;
+    CHECK_FAILURE(webView2->get_Profile(&profile));
+    CHECK_FAILURE(profile2->Delete());
+}
+
+void AppWindow::RegisterProfileDeletedEventHandlers(ICoreWebView2* webView2)
+{
+    wil::com_ptr<ICoreWebView2Profile> profile;
+    CHECK_FAILURE(webView2->get_Profile(&profile));
+    if (profile)
     {
-        wil::com_ptr<ICoreWebView2Profile> profile;
-        CHECK_FAILURE(webview7->get_Profile(&profile));
-        auto profile2 = profile.try_query<ICoreWebView2StagingProfile4>();
-        if (profile2)
-        {
-            CHECK_FAILURE(profile2->Delete());
-        }
+        CHECK_FAILURE(profile->add_Deleted(
```

Comment:
Event name is misleading, because the profile hasn't been deleted yet. It has merely been marked for deletion.

Maybe `DeletePending`? `DeleteStarting`?

How do you know when the deletion has completed, so it is safe to reuse the profile name?

Reply:
maybe OnDeletonRequested or OnMarkedForDeletion.  However it sounds like the profile is off-limits for any use before this event starts to be handled, in which case we may have picked "Deleted" to avoid suggesting it's in a zombie state that supports any valid interactions?

Reply:
When the `Profile.Deleted` event is raised, the deletion operation has executed partially. Any associated webview2 objects has been closed, but we can't completely delete local files because some files may still be occupied. So, we cannot know when the deletion has completed.
And if profile has only one instance, we do not need this event to tell the caller that the delete operation has completely. The reason for adding `Profile.Deleted` event is it may have multiple instances, they may in different process, so we must tell each instance about some instance has called `Profile.Deleted` and cannot use the profile and associated webview2.